src/d3a/d3a_core/sim_results/file_export_endpoints.py

```diff
@@ -52,7 +52,10 @@ def _get_stats_from_market_data(self, out_dict, area, balancing):
                 out_dict[area.slug][label].append(row[ii])
 
     def _populate_plots_stats_for_supply_demand_curve(self, area):
-        if ConstSettings.IAASettings.MARKET_TYPE == 3:
+        from d3a.d3a_core.singletons import bid_offer_matcher
+        if (ConstSettings.IAASettings.MARKET_TYPE == 2 and
+                ConstSettings.IAASettings.BID_OFFER_MATCH_TYPE ==
+                BidOfferMatchAlgoEnum.PAY_AS_CLEAR.value):
```

Comment:
```suggestion
        if (
            ConstSettings.IAASettings.MARKET_TYPE == SpotMarketTypeEnum.TWO_SIDED_PAY_AS_CLEAR.value 
            and ConstSettings.IAASettings.BID_OFFER_MATCH_TYPE == BidOfferMatchAlgoEnum.PAY_AS_CLEAR.value):
```

Reply:
I deliberately have avoided it as [this](https://github.com/gridsingularity/d3a-interface/pull/236) PR yet needs to be merged first